runtime-testsuite/test/org/antlr/v4/test/runtime/go/BaseTest.java

```diff
@@ -424,15 +428,21 @@ public String execModule(String fileName) {
 	private String locateTool(String tool) {
 		ArrayList<String> paths = new ArrayList<String>(); // default cap is about right
 
-		String pathEnv = System.getenv("PATH");
-		if (pathEnv != null) {
-			paths.addAll(Arrays.asList(pathEnv.split(pathSep)));
-		}
+		// GOROOT should have priority if set
 		String goroot = System.getenv("GOROOT");
 		if (goroot != null) {
 			paths.add(goroot + File.separatorChar + "bin");
 		}
 
+		String pathEnv = System.getenv("PATH");
+		if (pathEnv != null) {
+			paths.addAll(Arrays.asList(pathEnv.split(pathSep)));
+		}
+
+		// OS specific default locations of binary dist as last resort
+		paths.add("/usr/local/go/bin");
+		paths.add("c:\\Go\\bin");
+
```

Comment:
Won't it mess up programs that split GOPATH on ":"? There's going to be a "c" and a "\\Go\\bin" directory in GOPATH, albeit at the end.

> On May 28, 2016, at 1:37 AM, Wolfgang Johannes Kohnen <notifications@github.com> wrote:
> 
> In runtime-testsuite/test/org/antlr/v4/test/runtime/go/BaseTest.java:
> 
> >  		String goroot = System.getenv("GOROOT");
> >  		if (goroot != null) {
> >  			paths.add(goroot + File.separatorChar + "bin");
> >  		}
> >  
> > +		String pathEnv = System.getenv("PATH");
> > +		if (pathEnv != null) {
> > +			paths.addAll(Arrays.asList(pathEnv.split(pathSep)));
> > +		}
> > +
> > +		// OS specific default locations of binary dist as last resort
> > +		paths.add("/usr/local/go/bin");
> > +		paths.add("c:\\Go\\bin");
> > +
> Both : and \ are legal filename characters on UN*X:
> 
> wjk@empit:scarychars$ mkdir c:\\javastring
> wjk@empit:scarychars$ touch c:\\javastring/c:\\javastring
> wjk@empit:scarychars$ ls -lR
> .:
> total 4
> drwxr-xr-x 2 wjk wjk 4096 May 28 10:00 c:\javastring
> 
> ./c:\javastring:
> total 0
> -rw-r--r-- 1 wjk wjk 0 May 28 10:00 c:\javastring
> Even if the fs does not not support these chars (HFS+?), searching for such a path on the "wrong" filesystem does simply nothing. 
> 
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub, or mute the thread.
>